dask_ml/decomposition/pca.py

```diff
@@ -250,7 +250,9 @@ def _get_solver(self, X, n_components):
         return solver
 
     def _fit(self, X):
+        n_samples, n_features = X.shape
         if isinstance(X, dd.DataFrame):
+            n_samples = len(X)
```

Comment:
This will trigger a computation of the task graph backing the DataFrame. We'd like to avoid that until the main `compute` down around line 330.